exonum/src/blockchain/block.rs

```diff
@@ -62,6 +111,60 @@ pub struct Block {
     /// These calls can include transactions, `before_transactions` and/or `after_transactions` hooks
     /// for services.
     pub error_hash: Hash,
+    /// Additional information that can be added into the block.
+    pub entries: BlockHeaderEntries,
+}
+
+impl Block {
+    /// Insert new entry to the block header.
+    ///
+    /// # Usage
+    ///
+    /// ```ignore
+    /// use exonum::blockchain::{Block, BlockHeaderKey};
+    ///
+    /// impl BlockHeaderKey for ActiveServices {
+    ///     const NAME: &'static str = "ACTIVE_SERVICES";
+    ///     type Value = Self;
+    /// }
+    ///
+    /// let mut block = Block { ... };
+    ///
+    /// let services = ActiveServices::new();
+    /// block.insert::<ActiveServices>(services);
+    ///
+    /// ```
+    pub fn add_entry<K: BlockHeaderKey>(&mut self, value: K::Value) {
```

Comment:
Does this API need to be public? The fewer public APIs, the less probability to break them 🙃